mono/metadata/marshal.c

```diff
@@ -595,7 +595,9 @@ mono_delegate_free_ftnptr (MonoDelegate *delegate)
 		MonoMethod *method;
 
 		ji = mono_jit_info_table_find (mono_domain_get (), mono_get_addr_from_ftnptr (ptr));
-		g_assert (ji);
+		/* FIXME we leak wrapper with the interpreter */
+		if (!ji)
+			return;
```

Comment:
There is a `MonoJitDomainInfo::interp_code_hash` which has a `MonoMethod`->`InterpMethod` mapping. It's tempting to add the reverse...

Reply:
the delegate_trampoline is a code pointer that is used to invoke the delegate from native code. So it is a rgctx_tramp -> interp_in_wrapper -> n2m interp wrapper -> interp method. I think the interp entry wrappers are reused while the tramp and the wrapper should ideally be freed. Also I  don't see how we can easily go from the tramp code pointer to the n2m wrapper. I think we should store it in a new hash table for freeing.